pkg/template/api/validation/validation.go

```diff
@@ -12,15 +12,28 @@ import (
 
 var parameterNameExp = regexp.MustCompile(`^[a-zA-Z0-9\_]+$`)
 
+const (
+	// FIXME: There should be global constant specifying the maximum allowed
+	//				length for resources.
+	maxParameterNameLength        = 120
+	maxParameterDescriptionLength = 255
```

Comment:
You should validate parameters like annotations - total size matters more than individual length (name length is reasonable, everything else is arbitrary).